src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,135 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.InvalidRegistrationDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private User defaultUser;
+    private StorageDao storageDao;
+    private RegistrationService registrationService;
+
+    @BeforeEach
+    void setUp() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl();
+        defaultUser = new User();
+        defaultUser.setAge(90);
+        defaultUser.setLogin("initial_login");
+        defaultUser.setPassword("initial_password");
+    }
+
+    @Test
+    void register_nullArgument_notOk() {
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            registrationService.register(null);
+        }, "InvalidRegistrationDataException expected for null argument");
+    }
+
+    @Test
+    void register_loginIsNull_notOk() {
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setLogin(null);
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException expected for user.login = null");
+    }
+
+    @Test
+    void register_ageIsNull_notOk() {
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setLogin("first_login");
+            defaultUser.setAge(null);
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException expected for user.age = null");
+    }
+
+    @Test
+    void register_passwordIsNull_notOk() {
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setPassword(null);
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException expected for user.password = null");
+    }
+
+    @Test
+    void register_loginAlreadyInStorage_notOk() {
+        User storageUser = new User();
+        storageUser.setLogin("second_login");
+        storageDao.add(storageUser);
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setLogin("second_login");
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException, method allows user duplicates");
+    }
+
+    @Test
+    void register_loginIsEmpty_notOk() {
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setLogin("");
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException,"
+                + "when login field empty an exception should be thrown");
+    }
+
+    @Test
+    void register_default_ok() {
```

Comment:
missed the main thing, how can we know that the user was actually registered?
It should have an ID assigned